DOCSP-48420: bson v2.14 vector support (#169)
mongodb/docs-rust: source/fundamentals/aggregation/vector-search.txt

     - Description

   * - ``index``
     - string
     - Name of the vector search index

   * - ``path``
     - string
     - Field that contains vector embeddings

   * - ``queryVector``
     - array of numbers
     - Vector representation of your query

   * - ``limit``
     - number
     - Number of results to return

Atlas Search Query Examples
---------------------------

In this section, you can learn how to perform Atlas Vector
Search queries. The examples in this section use sample data from the
``sample_mflix.embedded_movies`` collection. To learn how to load this
sample data, see the :atlas:`Load Data into Atlas </sample-data/>`
tutorial in the Atlas documentation.

.. note:: Query Vector Length
    
   For demonstrative purposes, the examples in this section use
   sample query vectors that contain very few elements, compared to 
   the query vector you might use in a runnable application. To view an
   example that contains the full-length query vector, see the
   :atlas:`Atlas Vector Search Quick Start </atlas-vector-search/tutorials/vector-search-quick-start/>`
   and select :guilabel:`Rust` from the :guilabel:`Select your language` dropdown in the upper-right

>    corner of the page.

Reviewer: Q: Is the example still runnable without the correct query vector? If not, the output might be misleading.

Author: I'll add a code comment to direct users to replace the query vector

Reviewer: S: I know it looks ugly to have the long vector, but it might be worth adding it to the page rather than including a note explaining why you didn't include and how to get the longer vector. I'll leave up to your discretion!

   ``Vector`` type to improve storage efficiency.

Vector Search Score
~~~~~~~~~~~~~~~~~~~

The following code performs the same query as in the preceding example,
but outputs only the ``title`` field and ``vectorSearchScore`` meta
field, which describes how well the document matches the query vector:

.. io-code-block::
   :copyable: true

   .. input:: /includes/fundamentals/code-snippets/aggregation/vector-search.rs
      :language: rust
      :dedent:
      :start-after: start-score-query
      :end-before: end-score-query
      :emphasize-lines: 15

   .. output::
      :language: json
      :visible: false

      { "title": "Thrill Seekers", "score": 0.9332504272460938 }
      { "title": "About Time", "score": 0.9312690496444702 }
      { "title": "The Time Machine", "score": 0.9295310378074646 }
      { "title": "Crusade in Jeans", "score": 0.9290415048599243 }
      { "title": "Timecop", "score": 0.9283164739608765 }

Vector Search Options
---------------------

You can use a ``$vectorSearch`` stage to perform many types of Atlas
Vector Search queries. Depending on your desired query, you can specify the
following options in the stage definition:

> .. list-table::
>    :header-rows: 1

Reviewer: ```suggestion
.. list-table::
   :widths: 20 20 40 20
   :header-rows: 1
```
S: Edit column widths, so "document" doesn't wrap

   * - Optional Parameter
     - Type
     - Description
     - Default Value

   * - ``exact``
     - boolean
     - Specifies whether to run an Exact Nearest Neighbor (``true``) or
       Approximate Nearest Neighbor (``false``) search
     - ``false``

   * - ``filter``
     - document
     - Specifies a pre-filter for documents to search on

>      - No filtering

Author: This came from the Server docs, which didn't supply default values. I'm not sure `null` is actually the type interpreted by the server

>      - No filtering

Reviewer: Could be worth asking the tech reviewer for the default value/clarification

>      - No filtering

Reviewer 2: I'd leave these as-is; I'm not sure how the server represents these values when they're unset, but that shouldn't be of concern to the user and I think this phrasing better represents the actual behavior. There's no default value from the driver's perspective because the driver just sends the pipeline the user provided directly to the server.

   * - ``numCandidates``
     - number
     - Specifies the number of nearest neighbors to use during the
       search

>      - No limit

Reviewer: Would ask tech reviewer

To learn more about these parameters, see the :atlas:`Fields
</atlas-vector-search/vector-search-stage/#fields>` section of the

> ``$vectorSearch`` operator reference in the Atlas documentation.

Reviewer: Should you include an Additional Information section with API refs?

Vector Search Aggregation Stage
-------------------------------

To create a ``$vectorSearch`` stage in your aggregation pipeline, perform the
following actions:

1. Create a vector to store the pipeline stages.

#. Specify the ``$vectorSearch`` operator and provide details about the
   vector search query.

You must define the following fields in your ``$vectorSearch`` stage:

.. list-table::
   :header-rows: 1

   * - Parameter
     - Type
     - Description

   * - ``index``
     - string
     - Name of the vector search index

   * - ``path``
     - string
     - Field that contains vector embeddings

   * - ``queryVector``
     - array of numbers
     - Vector representation of your query

   * - ``limit``
     - number
     - Number of results to return

Atlas Search Query Examples
---------------------------

In this section, you can learn how to perform Atlas Vector
Search queries. The examples in this section use sample data from the
``sample_mflix.embedded_movies`` collection. To learn how to load this
sample data, see the :atlas:`Load Data into Atlas </sample-data/>`
tutorial in the Atlas documentation.

.. note:: Query Vector Length
    

>    For demonstrative purposes, the examples in this section use
>    sample query vectors that contain very few elements, compared to 
>    the query vector you use in a runnable application.

Reviewer: ```suggestion
   For demonstrative purposes, the example ``query_vector`` in this section 
   contains very few elements, compared to 
   the query vector you might use in a runnable application.
```
S: Just to make it clear what example vector you are referring to

Author: using the full query vector so removing this admonition
